### PR TITLE
Fix logcollector stop

### PR DESCRIPTION
**Pull request by TomasTurina**
|Related issue|
|---|
|https://github.com/wazuh/wazuh-agent/issues/450 https://github.com/wazuh/wazuh-agent/issues/449|

## Description

This PR includes the following fixes in log collector:
- Notify the coroutines to stop execution.
- Wait until coroutines terminate to clean readers.
- Cancel timer tasks to avoid agent process to hang in exit.

Besides, the default batch size is changed to 1MB.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
  - [x] MAC OS X

### CODE REVIEW

File: src/modules/logcollector/src/logcollector.cpp [hunk 1]
```diff
@@ -101,6 +118,40 @@ void Logcollector::AddReader(std::shared_ptr<IReader> reader) {
...
 Awaitable Logcollector::Wait(std::chrono::milliseconds ms) {
-    co_await boost::asio::steady_timer(m_ioContext, ms).async_wait(boost::asio::use_awaitable);
+    if (!m_ioContext.stopped()) {
+        auto timer = boost::asio::steady_timer(m_ioContext, ms);
+        {
+            std::lock_guard<std::mutex> lock(m_timersMutex);
+            m_timers.push_back(&timer);
+        }
+        try {
+            co_await timer.async_wait(boost::asio::use_awaitable);
```
Comment (jr0me) [inline src/modules/logcollector/src/logcollector.cpp:147]: If there are other reasons why this throws I'd change the message at least slightly so it is not misleading...

File: src/modules/logcollector/src/logcollector.cpp [hunk 2]
```diff
@@ -101,6 +118,40 @@ void Logcollector::AddReader(std::shared_ptr<IReader> reader) {
...
+
+    {
+        std::lock_guard<std::mutex> lock(m_timersMutex);
+        for (const auto &timer : m_timers) {
+            timer->cancel();
+        }
+    }
+
+    while (m_activeReaders) {
+        std::this_thread::sleep_for(std::chrono::milliseconds(10)); // NOLINT
```
Comment (jr0me) [inline src/modules/logcollector/src/logcollector.cpp:134]: We should specify which warning we're silencing in parenthesis

File: src/modules/logcollector/include/logcollector.hpp [hunk 3]
```diff
@@ -81,6 +82,9 @@ class Logcollector {
     /// @param configurationParser Configuration parser
     void SetupFileReader(const std::shared_ptr<const configuration::ConfigurationParser> configurationParser);
 
+    /// @brief Clean all readers
+    void CleanAllReaders();
```
Comment (jr0me) [inline src/modules/logcollector/include/logcollector.hpp:86]: Private?

**Review by jr0me — APPROVED** [review]

File: src/modules/logcollector/include/logcollector.hpp (hunk 3 above)
Comment (TomasTurina) [reply src/modules/logcollector/include/logcollector.hpp:86]: I will keep it this way as it could be useful in future usages.

**Review by aritosteles — APPROVED** [review]
LGTM!

**Review by Nicogp — APPROVED** [review]
LGTM!

**Pull request merged**